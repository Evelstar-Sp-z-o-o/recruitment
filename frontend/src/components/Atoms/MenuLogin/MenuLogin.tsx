import { useTranslation } from 'react-i18next';

import LoginIcon from '@mui/icons-material/Login';
import LogoutIcon from '@mui/icons-material/Logout';
import ListItemIcon from '@mui/material/ListItemIcon';
import ListItemText from '@mui/material/ListItemText';
import MenuItem from '@mui/material/MenuItem';

interface IMenuLoginProps {
  isLogin?: boolean;
  handleLogout?: () => void;
  handleToggleModal?: () => void;
}

const MenuLogin: FC<IMenuLoginProps> = ({ isLogin, handleLogout, handleToggleModal }) => {
  const { t } = useTranslation();

  return (
    <MenuItem sx={{ mt: '3rem' }} onClick={isLogin ? handleToggleModal : handleLogout}>
      <ListItemIcon>
        {isLogin ? (
          <LoginIcon data-testid="LoginIcon" fontSize="small" />
        ) : (
          <LogoutIcon data-testid="LogoutIcon" fontSize="small" />
        )}
      </ListItemIcon>
      <ListItemText>{isLogin ? t('menu.log.in') : t('menu.log.out')}</ListItemText>
    </MenuItem>
  );
};

export default MenuLogin;
