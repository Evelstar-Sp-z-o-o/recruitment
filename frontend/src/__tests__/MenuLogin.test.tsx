import { MemoryRouter } from 'react-router-dom';
import { create } from 'react-test-renderer';

import { vi } from 'vitest';

import MenuLogin from '@/src/components/Atoms/MenuLogin/MenuLogin';
import '@testing-library/jest-dom';
import { render, screen, fireEvent } from '@testing-library/react';

vi.mock('react-i18next', () => ({
  useTranslation: () => ({
    t: (key: string) => key,
  }),
}));

describe('MenuLogin Component', () => {
  it('renders correctly when logged in', () => {
    render(
      <MemoryRouter>
        <MenuLogin isLogin handleToggleModal={vi.fn()} />
      </MemoryRouter>,
    );

    expect(screen.getByRole('menuitem')).toBeInTheDocument();

    expect(screen.getByTestId('LoginIcon')).toBeInTheDocument();
    expect(screen.getByText('menu.log.in')).toBeInTheDocument();
  });

  it('renders correctly when logged out', () => {
    render(
      <MemoryRouter>
        <MenuLogin handleLogout={vi.fn()} />
      </MemoryRouter>,
    );

    expect(screen.getByRole('menuitem')).toBeInTheDocument();

    expect(screen.getByTestId('LogoutIcon')).toBeInTheDocument();
    expect(screen.getByText('menu.log.out')).toBeInTheDocument();
  });

  it('calls handleToggleModal when logged in and clicked', () => {
    const handleToggleModal = vi.fn();
    render(
      <MemoryRouter>
        <MenuLogin isLogin handleToggleModal={handleToggleModal} />
      </MemoryRouter>,
    );

    fireEvent.click(screen.getByRole('menuitem'));
    expect(handleToggleModal).toHaveBeenCalledTimes(1);
  });

  it('calls handleLogout when logged out and clicked', () => {
    const handleLogout = vi.fn();
    render(
      <MemoryRouter>
        <MenuLogin handleLogout={handleLogout} />
      </MemoryRouter>,
    );

    fireEvent.click(screen.getByRole('menuitem'));
    expect(handleLogout).toHaveBeenCalledTimes(1);
  });

  it('should match snapshot', () => {
    const tree = create(
      <MemoryRouter>
        <MenuLogin isLogin={true} handleToggleModal={vi.fn()} />
      </MemoryRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
