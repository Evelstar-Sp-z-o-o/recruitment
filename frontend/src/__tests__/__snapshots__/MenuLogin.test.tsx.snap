// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MenuLogin Component > should match snapshot 1`] = `
<li
  className="MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-root MuiMenuItem-gutters css-nhc2at-MuiButtonBase-root-MuiMenuItem-root"
  onBlur={[Function]}
  onClick={[MockFunction spy]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="menuitem"
  tabIndex={-1}
>
  <div
    className="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
      data-testid="LoginIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M11 7 9.6 8.4l2.6 2.6H2v2h10.2l-2.6 2.6L11 17l5-5-5-5zm9 12h-8v2h8c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-8v2h8v14z"
      />
    </svg>
  </div>
  <div
    className="MuiListItemText-root css-tlelie-MuiListItemText-root"
  >
    <span
      className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
    >
      menu.log.in
    </span>
  </div>
</li>
`;
